### ISSUE TITLE
"authenticate" module does not handle connection timeout error

### ISSUE THREAD

**Issue by bdabak** [ArnaudBuchholz/node-ui5#7]
Hi,

await authenticate(CONNECTION) causes times out and is not handled. 
Therefore application waits forever.

**Comment by ArnaudBuchholz** [comment]
Indeed, I will provide you a working version but it can't be part of node-ui5 since it is in the middle of a refactoring...

**Comment by bdabak** [comment]
Thanks I will be waiting

**Comment by ArnaudBuchholz** [comment]
Hi @bdabak, can you please get the latest version of [basic-with-csrf.js](https://github.com/ArnaudBuchholz/node-ui5/blob/master/lib/authenticate/basic-with-csrf.js) and replace it inside your node_modules\node-ui5\lib folder?
